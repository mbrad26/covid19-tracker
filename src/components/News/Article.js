import React from 'react'
import { Card } from 'react-bootstrap';

const Article = ({ article }) => {

  return (
      <Card id='article'>
        <Card.Img variant="top" src={article.urlToImage} />
        <Card.Body>
          <Card.Title>{article.title}</Card.Title>
          <Card.Text id='source'><span>{article.source.name}</span></Card.Text>
          <Card.Text className='text-muted far fa-clock'>
            <span> {new Date(article.publishedAt).toDateString()}</span>
          </Card.Text>
          <Card.Text id='article-description'>{article.description}</Card.Text>
          <a href={article.url} target='_blank' rel='noopener noreferrer'>Read full article ...</a>
        </Card.Body>
      </Card>
  );
};

export default Article;
