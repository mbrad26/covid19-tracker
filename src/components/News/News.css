.news {
  overflow-y: scroll;
  height: 81vh;
}

#spinner-news {
  margin: 100px;
}

#article {
  margin-bottom: 10px;
  background-color: #E9E8E6;
}

#article-description {
  padding: 5px 5px 0px 0px;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

#source {
  margin-bottom: 0px;
}

.far.fa-clock {
  font-size: 0.8rem;
  margin-bottom: 10px;
}
