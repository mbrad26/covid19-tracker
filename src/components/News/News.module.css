.card-img-top {
  min-width: 250px;
  height: auto;
}
