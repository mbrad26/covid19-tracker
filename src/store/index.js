import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
