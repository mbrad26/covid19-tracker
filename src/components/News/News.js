import React, { useEffect } from 'react';
import { Spinner, Row, Col } from 'react-bootstrap';

import './News.css';
import Article from './Article';

const News = ({
  isError,
  articles,
  isLoading,
  loadingNewsData,
  }) => {

  useEffect(() => {
    loadingNewsData();
  }, [loadingNewsData]);

  console.log('NEWS: ', isLoading);

  return (
    <div className='news'>
      {isLoading && <Spinner id='spinner-news' animation="border" variant="light" role="status"/>}
      {articles &&
        articles.slice(0, 8).map(article =>
          <Article key={article.url} article={article} />
        )
      }
      {isError && <h3>Something went wrong ...</h3>}
    </div>
  );
};

export default News;
