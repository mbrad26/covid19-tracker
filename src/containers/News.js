import { connect } from 'react-redux';

import News from '../components/News/News';
import { doNewsLoading } from '../actions/newsData';
import { getNewsData, getIsLoadingStatus, getIsErrorStatus } from '../selectors/data';

const mapStateToProps = ({ newsDataState }) => ({
  articles: getNewsData(newsDataState),
  isError: getIsErrorStatus(newsDataState),
  isLoading: getIsLoadingStatus(newsDataState),
});

const mapDispatchToProps = dispatch => ({
  loadingNewsData: () => dispatch(doNewsLoading()),
});

export default connect(mapStateToProps, mapDispatchToProps)(News);
