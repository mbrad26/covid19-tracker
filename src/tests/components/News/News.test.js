import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../store';
import News from '../../../containers/News';

describe('News', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store} >
        <News />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
