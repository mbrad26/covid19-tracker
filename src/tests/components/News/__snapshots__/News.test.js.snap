// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News renders snapshot 1`] = `
<div
  class="news row"
>
  <div
    class="spinner-border text-light"
    id="spinner-news"
    role="status"
  />
</div>
`;
